XenModel/Actions/USB/DeleteVUSBAction.cs

```diff
@@ -52,15 +52,20 @@ protected override void Run()
         {
             try
             {
-                VUSB.async_unplug(Session, _vusb.opaque_ref);
-                PollToCompletion(0, 50);
+                if ((_vusb.Connection.Resolve(_vusb.attached) != null) &&
+                    XenAPI.VUSB.get_allowed_operations(Session, _vusb.opaque_ref).Contains(XenAPI.vusb_operations.unplug))
+                {
+                    VUSB.async_unplug(Session, _vusb.opaque_ref);

```

Comment:
PollToCompletion relies on the action's `RelatedTask` property being assigned, so it can poll that task. 
So we need to assign it like `RelatedTask = VUSB.async_unplug(Session, _vusb.opaque_ref);`

Same for the async_destroy call below.

Reply:
Thank you, I saw DetachVirtualDiskAction has similar usage.